test/validator.js

```diff
@@ -508,6 +508,47 @@ describe('Validator', () => {
                 }
             });
         });
+
+        it('reports warnings with nested options', () => {
```

Comment:
We'll definitely want more tests here to cover root level unknown as well as `allowUnknown: 'forbidden'`